components/style/rule_tree/mod.rs

```diff
@@ -801,17 +814,22 @@ impl Drop for StrongRuleNode {
         let root = unsafe { &*node.root.as_ref().unwrap().ptr() };
         let free_list = &root.next_free;
 
-        // We're sure we're already in the free list, don't spinloop.
-        if node.next_free.load(Ordering::SeqCst) != ptr::null_mut() {
+        // We're sure we're already in the free list, don't spinloop if we're.
+        // Note that this is just a fast path, so it doesn't need to have an
+        // strong memory ordering.
+        if node.next_free.load(Ordering::Relaxed) != ptr::null_mut() {
             return;
         }
 
         // Ensure we "lock" the free list head swapping it with a null pointer.
-        let mut old_head = free_list.load(Ordering::SeqCst);
+        //
+        // Note that we can use relaxed operations, we'll fail the
+        // compare_exchange_weak anyway if we fail.
```

Comment:
This comment needs updating. Worth explaining in some detail why (Acquired, Relaxed) is the right ordering here.